source/simulator/core.cc

```diff
@@ -142,11 +142,7 @@ namespace aspect
     melt_handler (parameters.include_melt_transport ?
                   std_cxx14::make_unique<MeltHandler<dim>>(prm) :
                   nullptr),
-    newton_handler ((parameters.nonlinear_solver == NonlinearSolver::iterated_Advection_and_Newton_Stokes ||
-                     parameters.nonlinear_solver == NonlinearSolver::single_Advection_iterated_Newton_Stokes ||
-                     parameters.nonlinear_solver == NonlinearSolver::no_Advection_iterated_defect_correction_Stokes ||
-                     parameters.nonlinear_solver == NonlinearSolver::single_Advection_iterated_defect_correction_Stokes ||
-                     parameters.nonlinear_solver == NonlinearSolver::iterated_Advection_and_defect_correction_Stokes) ?
+    newton_handler (Parameters<dim>::is_defect_correction(parameters.nonlinear_solver) ?
```

Comment:
Here, the property seems to be "requires NewtonHandler". That would suggest calling the function `requires_newton_handler()`.

That makes this suspicious, though -- why would `no_Advection_iterated_defect_correction_Stokes` require a `NewtonHandler`?

Reply:
Why does that make you suspicious? They are separate things. A model with no advection can still be very nonlinear (for the first timestep).

Reply:
Right. What I meant to say is this: there is a disconnect somewhere if we initialize a *Newton* handler also in case where we don't actually do a Newton iteration (e.g., `no_Advection_and_defect_correction_Stokes`). It may be that the `NewtonHandler` class also provides services for other solvers, in which case it is probably that that class is misnamed and `is_defect_correction()` is the correct name. All I meant to suggest is that one or the other is misnamed -- that the juxtaposition of the two names is dissonant and suggests that one of the two is misnamed.

Reply:
Ah, then I totally agree. The name of the NewtonHandler currently does not reflect anymore what services it actually provides.